### DISCUSSION TITLE
Inline Popup Editor for Text Edits

### DISCUSSION THREAD

**Discussion by pistonsfan2004** [logancyang/obsidian-copilot#963] [Ideas]
A popup editor triggered by text selection would be super helpful. It could let users quickly instruct the model to reword, expand, shorten, or refine the selected text through a small chat-like field. You could either type custom instructions or pick from predefined commands (like with the slash command menu). The full note would be used as context to keep edits cohesive, and you’d have the option to preview, accept, or reject changes—all without needing to rely on the sidebar.

**Comment by pistonsfan2004** [comment]
like this, but natively integrated with copilot

<img width="1206" alt="image" src="https://github.com/user-attachments/assets/148089fc-8058-42e7-b8da-2003c89206f6" />

**Reply by logancyang** [reply]
This is exactly what I have in mind, thanks for the suggestion!

Inline features will be the next big focus of Copilot, stay tuned! cc @zeroliu @Emt-lin
